mlrun/api/utils/clients/chief.py

```diff
@@ -0,0 +1,75 @@
+import copy
+
+import requests.adapters
+import urllib3
+
+import mlrun.api.schemas
+import mlrun.api.utils.projects.remotes.follower
+import mlrun.errors
+import mlrun.utils.singleton
+from mlrun.utils import logger
+
+
+class Client(
+    metaclass=mlrun.utils.singleton.AbstractSingleton,
+):
+    """
+    We chose chief-workers architecture to provide multi-instance API.
+    By default, all API calls can access both the chief and workers.
+    The key distinction is that some responsibilities, such as scheduling jobs, are exclusively performed by the chief.
+    Instead of limiting the ui/client to only send requests to the chief, which would cause the entire scaling solution
+    to be redundant.
+    When one of the workers receives a request that the chief needs to execute or may have the knowledge of that piece
+    of information, the worker will redirect the request to the chief.
```

Comment:
"which would cause the entire scaling solution to be redundant" I guess you meant doing such thing will hurt the performance increase we're getting from the multi-instance API - but that's not the reason for redirecting, when you're redirecting you're practically redirecting the load to the chief, which means we exactly as you said, hurting the performance gain from this feature, therefore we want the chief to have the minimum possible special responsibilities.
The reason for redirecting is that we don't want (cause there's no sense) clients to have to "know" which responsibilities the chief has